diff --git a/src/main/scala/io/iohk/atala/prism/node/services/CardanoLedgerService.scala b/src/main/scala/io/iohk/atala/prism/node/services/CardanoLedgerService.scala
index 8ca7f6055..79e6c0ead 100644
--- a/src/main/scala/io/iohk/atala/prism/node/services/CardanoLedgerService.scala
+++ b/src/main/scala/io/iohk/atala/prism/node/services/CardanoLedgerService.scala
@@ -90,7 +90,7 @@ class CardanoLedgerService[F[_]] private[services] (
   }
 
   // Schedule the initial sync
-  scheduleSync(30.seconds)
+  scheduleSync(5.seconds)
 
   /** Publishes AtalaObject containing a list of operations inside.
     */
@@ -148,7 +148,7 @@ class CardanoLedgerService[F[_]] private[services] (
               // There blocks to sync, don't wait to sync faster
               scheduleSync(0.seconds)
             } else {
-              scheduleSync(20.seconds)
+              scheduleSync(1.seconds)
             }
           }
     )
