diff --git a/build.sbt b/build.sbt
index fb12ce4b..5611a866 100644
--- a/build.sbt
+++ b/build.sbt
@@ -439,31 +439,7 @@ val commonSetttings = Seq(
   // Needed for Kotlin coroutines that support new memory management mode
   resolvers += "JetBrains Space Maven Repository" at "https://maven.pkg.jetbrains.space/public/p/kotlinx-coroutines/maven",
   // Needed for com.github.multiformats:java-multibase
-  resolvers += "jitpack" at "https://jitpack.io",
-  // Override 'updateLicenses' for all project to inject custom DependencyResolution.
-  // https://github.com/sbt/sbt-license-report/blob/9675cedb19c794de1119cbcf46a255fc8dcd5d4e/src/main/scala/sbtlicensereport/SbtLicenseReport.scala#L84
-  updateLicenses := {
-    import sbt.librarymanagement.DependencyResolution
-    import sbt.librarymanagement.ivy.IvyDependencyResolution
-    import sbtlicensereport.license
-
-    val ignore = update.value
-    val overrides = licenseOverrides.value.lift
-    val depExclusions = licenseDepExclusions.value.lift
-    val originatingModule = DepModuleInfo(organization.value, name.value, version.value)
-    val resolution =
-      DependencyResolution(new LicenseReportCustomDependencyResolution(ivyConfiguration.value, ivyModule.value))
-    license.LicenseReport.makeReport(
-      ivyModule.value,
-      resolution,
-      licenseConfigurations.value,
-      licenseSelection.value,
-      overrides,
-      depExclusions,
-      originatingModule,
-      streams.value.log
-    )
-  }
+  resolvers += "jitpack" at "https://jitpack.io"
 )
 
 lazy val commonConfigure: Project => Project = _.settings(
diff --git a/project/build.sbt b/project/build.sbt
deleted file mode 100644
index d0e56635..00000000
--- a/project/build.sbt
+++ /dev/null
@@ -1,7 +0,0 @@
-// Using unreleased plugin from 'main' which accepts DependencyResolution interface for building license report.
-lazy val sbtLicenseReportPlugin = ProjectRef(
-  uri("https://github.com/sbt/sbt-license-report.git#9675cedb19c794de1119cbcf46a255fc8dcd5d4e"),
-  "sbt-license-report"
-)
-
-lazy val root = (project in file(".")).dependsOn(sbtLicenseReportPlugin)
